borg/helpers.py

```diff
@@ -885,9 +891,9 @@ def canonical_path(self):
             return self.path
         else:
             if self.path and self.path.startswith('~'):
-                path = '/' + self.path
+                path = '/' + self.path  # /~/x = path x relative to home dir
             elif self.path and not self.path.startswith('/'):
-                path = '/~/' + self.path
+                path = '/./' + self.path  # /./x = path x relative to cwd
```

Comment:
before this change, this was a bit strange: a relative path was transformed to a homedir-relative url.
as we support generic relative urls now, we transform it into such a url.